app/main.py

```diff
@@ -1 +1,39 @@
-# write your code here
+import os
+
+
+class ReadAndDelete:
+    def __init__(self, file_name: str) -> None:
+        self.filename = file_name
+
+    def __enter__(self):
+        self.file = open(self.filename, "r")
+        return self.file
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if os.path.exists(self.filename):
```

Comment:
I think you don`t need this "if " statement, because if this method runs - the file already exists because it`s a method of the file